src/main/java/com/google/devtools/build/lib/rules/proto/ProtoLangToolchain.java

```diff
@@ -44,8 +44,12 @@ public ConfiguredTarget create(RuleContext ruleContext)
       ProtoInfo protoInfo = protos.get(ProtoInfo.PROVIDER);
       // TODO(cushon): it would be nice to make this mandatory and stop adding files to build too
       if (protoInfo != null) {
-        blacklistedProtos.addTransitive(protoInfo.getTransitiveProtoSources());
+        // TODO(yannic): Switch to getTransitiveProtoSources() and remove references to
+        // original proto sources when ProtoInfo is mandatory.
+        blacklistedProtos.addAll(protoInfo.getOriginalDirectProtoSources());
```

Comment:
Using direct instead of transitive sources un-does a change made in 01df1e5c3aaf709e9388293514ae978e59ae2cbb that I think is going to break some uses of `proto_lang_toolchain.blacklisted_protos`.

Reply:
Yep, this is exactly what I found out after debugging the aforementioned test failures. Adding _both_ here isn't very nice so I'll try adding the set of transitive blacklisted sources (I'd totally ask Yannic to do so, but since the tests that break are internal, he'd be flying blind...)

Reply:
...and it blends! I'm sending this out (+my changes) for internal review.